File: backend/entity/notification.go
package entity

import "gorm.io/gorm"

type Notification struct {
	gorm.Model
	Title   string `json:"title"`
	Type    string `json:"type"`    // เช่น "report_reply", "refund", "system"
	Message string `json:"message"`

	UserID uint  `json:"user_id"`
	User   *User `gorm:"foreignKey:UserID" json:"user,omitempty"`

	IsRead bool `json:"is_read" gorm:"default:false"`

	// ✅ อ้างอิงคำร้อง เพื่อให้หน้า UI เปิดดูรายละเอียด/ไฟล์แนบได้
	ReportID *uint          `json:"report_id"`
	Report   *ProblemReport `gorm:"foreignKey:ReportID" json:"report,omitempty"`
}

File: frontend/src/interfaces/Notification.ts
// src/interfaces/Notification.ts
import type { User } from "./User";
import type { ProblemReport } from "./problem_report";

export interface Notification {
  ID: number;

  title: string;
  message: string;
  type: string;          // e.g. "report_reply", "refund", "system"

  user_id: number;
  user?: User;

  is_read: boolean;      // ให้เป็น boolean ชัดเจน (ไม่ต้อง optional)
  created_at?: string;
  updated_at?: string;

  // ✅ อ้างอิงไปยังคำร้อง/รีพอร์ต เพื่อเปิดดูข้อความ/ไฟล์แนบใน UI
  report_id?: number;
  report?: ProblemReport;
}

export interface CreateNotificationRequest {
  title: string;
  type: string;
  message: string;
  user_id: number;

  // ✅ ระบุ report ที่เกี่ยวข้อง (ถ้ามี)
  report_id?: number;
}

export interface UpdateNotificationRequest {
  ID: number;
  title?: string;
  type?: string;
  message?: string;
}